assets/js/base/components/cart-checkout/address-form/test/index.js

```diff
@@ -162,17 +159,11 @@ describe( 'AddressForm Component', () => {
 		inputAddress( secondaryAddress );
 		// Only update `country` to verify other values are reset.
 		inputAddress( { country: primaryAddress.country } );
-
-		expect( screen.getByLabelText( cityRegExp ).value ).toBe( '' );
 		expect( screen.getByLabelText( stateRegExp ).value ).toBe( '' );
-		expect( screen.getByLabelText( postalCodeRegExp ).value ).toBe( '' );
```

Comment:
I thought the expected behavior was that when the country changed, all other address fields would be reset. That's especially important for countries which don't have a state or postal code, so we don't carry over values from the previous address. Was this behavior changed on purpose?

Reply:
Yeah, I will check this again, but when you're filling out the form, if fields clear because you input country last that's super annoying. It also breaks autocomplete. Ideally (and what I think I checked) fields should be cleared in the background if they are not needed. We don't need to do this in client/form.

Reply:
@Aljullu the new logic is here https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4369/files#diff-54ee17137efc6b9816bcbf193608f7a40333290f7deee587370a1dbcad331f9fR140-R147 It will clear the value if the field is hidden for the locale. Otherwise, there is no need to clear.

Reply:
> Yeah, I will check this again, but when you're filling out the form, if fields clear because you input country last that's super annoying. It also breaks autocomplete. Ideally (and what I think I checked) fields should be cleared in the background if they are not needed. We don't need to do this in client/form.

Makes sense. :ok_hand: